edx_when/api.py

```diff
@@ -398,11 +398,29 @@ def get_schedules_with_due_date(course_id, assignment_date):
     # Using those relative dates, get all Schedules that have a "hit" by working backwards to the start_date
     rel_start_dates = [assignment_date - rel_date for rel_date in rel_dates]
 
-    # Exclude any user that has an overridden date for a course on the specified day so there aren't duplicates
-    return Schedule.objects.filter(
-        enrollment__course_id=course_id,
-        start_date__date__in=rel_start_dates,
-    ).exclude(enrollment__user_id__in=user_ids).select_related('enrollment') | user_date_overridden_schedules
+    if rel_start_dates:
+        # Exclude any user that has an overridden date for a course on the specified day so there aren't duplicates
+        schedules = Schedule.objects.filter(
+            enrollment__course_id=course_id,
+            enrollment__is_active=True,
+            start_date__date__in=rel_start_dates,
+        ).exclude(enrollment__user_id__in=user_ids).select_related('enrollment') | schedules
+
+    # Add in all users with relative dates to exclude from the absolute dates query to prevent duplicates
+    user_ids = schedules.all().values_list('enrollment__user_id', flat=True).distinct()
+
+    has_abs_date_on_day = models.ContentDate.objects.filter(
+        course_id=course_id,
+        policy__abs_date__date=assignment_date
+    ).first()
+
+    if has_abs_date_on_day:
+        schedules = Schedule.objects.filter(
+            enrollment__course_id=course_id,
+            enrollment__is_active=True,
+        ).exclude(enrollment__user_id__in=user_ids).select_related('enrollment') | schedules
```

Comment:
Wait, I don't see anything that is restricting to the day here. Am I misunderstanding how this works?

Reply:
Yeah so for absolute days, everyone that's enrolled will have that same absolute day, so you'll, if they have a UserDate override they'll already be omitted from this result in the .exclude(), else everyone else who's active and enrolled should get the email, at least that was my thinking behind it.

Reply:
Hmm, OK. I understand this now. Can you add maybe a comment trying to explain that? To someone like me that hasn't been in this code for a bit, I wasn't thinking clearly about this and the "if A, then Z" stuff here felt like a logical jump. Though it's not.